lib/processor.js

```diff
@@ -58,14 +61,23 @@ function preprocess(text) {
 }
 
 /**
+ * Excludes unsatisfiable rules from the list of messages.
+ * @param {Message} message A message from the linter.
+ * @returns {boolean} True if the message should be included in output.
+ */
+function excludeUnsatisfiableRules(message) {
+    return UNSATISFIABLE_RULES.indexOf(message.ruleId) < 0;
```

Comment:
This is definitely something we can do as soon as core drops support for Node < 4.